### ISSUE TITLE
Support for APA102 (Dotstar)?

### ISSUE THREAD

**Issue by terrag42** [chrismelba/lightclockwifiv2#1]
I'm excited about building this project, but I happen to have a 1 meter strip of 144/m APA102 LEDs, not Neopixels. Is there a modification I can make to the code that will let me run this type of addressable LED?

**Comment by chrismelba** [comment]
I don't know what specific libraries can run those LEDs on ESP8266. I
imagine FastLED would be able to, but they haven't got the code working on
ESP8266 arduino yet.

https://github.com/FastLED/FastLED/issues/119

If you find a compatible library for your LEDs please let me know :)

**Comment by mager33** [comment]
FastLED works fine with the AP102 (I just gave it a try :-)

**Comment by chrismelba** [comment]
Awesome! Did you use the fast led build for esP8266? I've not had a chance
to try it yet.

**Comment by mager33** [comment]
Yes, and it was totally straightforward: downloard library, place in
Ardunino's lib-folder, start Arduino IDE, choose example, adjust pins,
number of LEDs and chiptype -> great show! PS: some of the examples have
to be adjusted slightly to accept a data pin AND a clock pin, but that
is easy!

**Comment by chrismelba** [comment]
I'm having a lot of trouble getting the FastLED library to work. I keep
getting errors saying it doesn't support extensa architecture? Have I got
the wrong library?

**Comment by mkeyno** [comment]
@chrismelba  it is work now , I have same project to store image data in SPIFF and then injected to APA102 LED strip , however after couple of try I find current version of FastLED not yet supported hardware SPI for ESP and hence it is not fast as it should be , now I'm trying to find the repo that could support HW SPI for APA102 chip
